File: core/src/main/java/com/neqsoft/reso/info/InfoActivity.java
package com.neqsoft.reso.info;

import android.os.Bundle;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.neqsoft.reso.R;
import com.neqsoft.reso.device.DeviceFragment;
import com.neqsoft.reso.os.OsFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class InfoActivity extends AppCompatActivity {

  @Nullable
  private Fragment deviceFragment, osFragment;
  private final FragmentManager fragmentManager;

  public InfoActivity() {
    fragmentManager = getSupportFragmentManager();
  }

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_info);
    setSupportActionBar(findViewById(R.id.toolbar));
    setupBottomNavigation();
    displayDeviceView();
  }

  private void setupBottomNavigation() {
    BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNavigationView);
    bottomNavigationView.setOnNavigationItemSelectedListener(menuItem -> {
      switch (menuItem.getItemId()) {
        case R.id.deviceItem:
          displayDeviceView();
          return true;
        case R.id.osItem:
          displayOsView();
          return true;
        case R.id.cpuItem:
          displayCpuView();
          return true;
        default:
          return false;
      }
    });
  }

  private void displayCpuView() {

  }

  private void displayOsView() {
    osFragment = osFragment != null ? osFragment : new OsFragment();
    display(osFragment);
  }

  private void displayDeviceView() {
    deviceFragment = deviceFragment != null ? deviceFragment : new DeviceFragment();
    display(deviceFragment);
  }

  private void display(@NonNull final Fragment fragment) {
    FragmentTransaction transaction = fragmentManager.beginTransaction();
    transaction.replace(R.id.container, fragment).commit();
  }
}

File: core/src/main/java/com/neqsoft/reso/info/InfoActivity.kt
package com.neqsoft.reso.info

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.neqsoft.reso.R
import com.neqsoft.reso.cpu.CpuInfoFragment
import com.neqsoft.reso.device.DeviceFragment
import com.neqsoft.reso.os.OsFragment

class InfoActivity : AppCompatActivity() {

    private val fragmentManager: FragmentManager = supportFragmentManager
    private val deviceFragment: Fragment by lazy { DeviceFragment() }
    private val osFragment: Fragment by lazy { OsFragment() }
    private val cputFragment: Fragment by lazy { CpuInfoFragment() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_info)
        setSupportActionBar(findViewById(R.id.toolbar))
        setupBottomNavigation()
        display(deviceFragment)
    }

    private fun setupBottomNavigation() {
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottomNavigationView)
        bottomNavigationView.setOnNavigationItemSelectedListener { menuItem: MenuItem ->
            when (menuItem.itemId) {
                R.id.deviceItem -> {
                    display(deviceFragment)
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.osItem -> {
                    display(osFragment)
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.cpuItem -> {
                    display(cputFragment)
                    return@setOnNavigationItemSelectedListener true
                }
                else -> return@setOnNavigationItemSelectedListener false
            }
        }
    }

    private fun display(fragment: Fragment) {
        val transaction = fragmentManager.beginTransaction()
        transaction.replace(R.id.container, fragment).commit()
    }
}